packages/select/src/Select.tsx

```diff
@@ -118,16 +119,11 @@ export default function Select({
     }
   }, [onChange, readOnly, value]);
 
-  if (!label && !ariaLabelledBy) {
-    console.error(
-      'For screen-reader accessibility, label or aria-labelledby must be provided to IconButton.',
-    );
-  }
+  validateLabelProps({ label, 'aria-labelledby': ariaLabelledBy }, 'Select');
```

Comment:
For example, we use the label and arialabelledby props in the component here, so we need to have them separated from rest

Reply:
Then we should do this check before destructuring the props, right?

Reply:
as long as it is at least checking for the presence of one or the other of the two props I don't see an issue

Reply:
My main issue with this is that we aren't actually validating that `aria-labelledby` is spelled correctly, for example (something typescript doesn't check for, though our a11y eslint plugin does in certain cases). We're not testing that the key exists in props, which is one of the reasons for the function.

My point here is, we're only using this function to half of its potential, and also increasing the maintenance burden by setting the keys that we're passing to the function manually in each component.